### PR TITLE
New features: Multiple references and more

### PR DESCRIPTION
**Pull request by baytelman**
- Multiple references: `["reference.displayField"]`
- Add optional `label` for each attribute
- `Date` fields (with `showTime` option)

### CODE REVIEW

File: demo/src/index.tsx [hunk 1]
```diff
@@ -8,9 +8,32 @@ import fakeDataProvider from 'ra-data-fakerest';
 
 const dataProvider = fakeDataProvider({
   posts: [
-    { id: 0, author: 10, title: 'Hello, world!', views: 123, tags: ['1', '5', '10'] },
-    { id: 1, author: 11, title: 'FooBar', views: 56789, tags: [] },
-    { id: 2, author: 10, title: 'Goodbye, all.', views: 777, tags: ['5'] }
+    {
+      id: 0,
+      date: new Date('2018-09-01T12:34:56'),
```
Comment (baytelman) [inline demo/src/index.tsx]: We support Date as native objects...

File: demo/src/index.tsx [hunk 2]
```diff
@@ -8,9 +8,32 @@ import fakeDataProvider from 'ra-data-fakerest';
...
+      author: 10,
+      readers: [11],
+      title: 'Hello, world!',
+      views: 123,
+      tags: ['1', '5', '10']
+    },
+    {
+      id: 1,
+      date: '2018-09-06T12:34:56',
```
Comment (baytelman) [inline demo/src/index.tsx]: ...or strings.

File: demo/src/index.tsx [hunk 3]
```diff
@@ -38,13 +66,26 @@ enum Tags {
 }
 
 const POST_SCHEMA = [
+  {
+    attribute: 'date',
+    type: Date,
```
Comment (baytelman) [inline demo/src/index.tsx]: New field type

File: demo/src/index.tsx [hunk 4]
```diff
@@ -38,13 +66,26 @@ enum Tags {
...
+    showTime: true,
```
Comment (baytelman) [inline demo/src/index.tsx]: with optional `showTime`

File: demo/src/index.tsx [hunk 5]
```diff
@@ -38,13 +66,26 @@ enum Tags {
...
+    label: 'Creation'
+  },
   {
     attribute: 'author',
     type: 'users.name'
   },
+  {
+    attribute: 'readers',
+    type: ['users.name'],
```
Comment (baytelman) [inline demo/src/index.tsx]: Multiple external references

File: lib/AutoAdmin.d.ts [hunk 6]
```diff
@@ -1,9 +1,11 @@
 /// <reference types="react" />
 interface AutoAdminAttribute {
     attribute: string;
-    type: string | Object | NumberConstructor | StringConstructor | AutoAdminAttribute[];
+    type: string | string[] | Object | DateConstructor | NumberConstructor | StringConstructor | AutoAdminAttribute[];
```
Comment (nschurmann) [inline lib/AutoAdmin.d.ts]: ❤️ yay!, typescript!

**Comment by nschurmann** [comment]
Why are we including autogenerated files in the commit?

**Comment by baytelman** [comment]
@nschurmann I don't have best practices for auto-generated files being included in the commit? Is this something usually done directly on master?

**Pull request merged**